frame/support/procedural/src/pallet/expand/hooks.rs

```diff
@@ -251,11 +251,16 @@ pub fn expand_hooks(def: &mut Def) -> proc_macro2::TokenStream {
 			for #pallet_ident<#type_use_gen> #where_clause
 			{
 				fn integrity_test() {
-					<
-						Self as #frame_support::traits::Hooks<
-						<T as #frame_system::Config>::BlockNumber
-						>
+					let ext = #frame_support::sp_io::TestExternalities::default();
+					let mut ro_ext = #frame_support::private::ReadOnlyExternalities::from(&ext.backend);
```

Comment:
Why read only?

Reply:
My thinking was that this should make developers aware that storage changes are being discarded. Or do you think its not needed?

Reply:
I mean  this is a test as the name says. Maybe we should improve the docs. However, there could be someone calling a "real" function for the test that also writes something.